site/public/js/docker_interface.js

```diff
@@ -17,3 +17,114 @@ function collapseSection(id,btn_id) {
         btn.innerHTML = 'Collapse';
     }
 }
+
+function filterOnClick() {
+    const this_filter = $(this).data('capability');
+
+    $('.filter-buttons').each(function (){
+        $(this).addClass('fully-transparent');
+    });
+
+    if ($(this).hasClass('fully-transparent')) {
+        $(this).removeClass('fully-transparent');
+    }
+
+    $('.image-row').each(function() {
+        const this_row = $(this);
+        let hide = true;
+        $(this).find('.badge').each(function (){
+            if ($(this).text() == this_filter) {
+                hide = false;
+            }
+        });
+        if (hide) {
+            this_row.hide();
+        }
+        else {
+            this_row.show();
+        }
+    });
+}
+
+function showAll() {
+    $('.image-row').show();
+    $('.filter-buttons').removeClass('fully-transparent');
+}
+
+function addFieldOnChange() {
+    const command = $(this).val();
+    const regex = new RegExp('^[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+/[a-z0-9]+[a-z0-9._(__)-]*[a-z0-9]+:[a-zA-Z0-9][a-zA-Z0-9._-]{0,127}$');
+    if (!regex.test(command)) {
+        $('#send-button').attr('disabled',true);
+        if (command !== '') {
+            $('#docker-warning').css('display', '');
+        }
+        else {
+            localStorage.removeItem('capability');
+        }
+    }
+    else {
+        $('#send-button').attr('disabled',false);
+        $('#docker-warning').css('display', 'none');
+        localStorage.setItem('capability', command);
+    }
+}
+
+function addImage(url) {
+    const capability = $('#capability-form').val();
+    const image = $('#add-field').val();
+    $.ajax({
+        url: url,
+        type: 'POST',
+        data: {
+            'capability': capability,
+            'image': image,
+            // eslint-disable-next-line no-undef
+            csrf_token: csrfToken,
+        },
+        success: function(data) {
+            const json = JSON.parse(data);
+            if (json.status == 'success') {
+                localStorage.removeItem('capability');
+            }
+            else {
+                window.alert('Something went wrong. Please try again.');
+            }
+            location.reload();
```

Comment:
I think having a location.reload at the end of a ajax call defeats the purpose of async JS, this could just be a native HTML form pointing to the addImage endpoint

Reply:
The `location.reload` was left there to show the messages properly. For some reason the success/error messages doesn't show up properly for me on the site without reloading the page.

Reply:
If the only reason is to display those messages and not update state then you can just pass the information back in the `JsonResponse` and use the JS `displaySuccessMessage` and `displayErrorMessage` to avoid the refresh.

The addErrorMessage on the PHP side will queue the message for the next page the user visits, its by design since the typical use would be to send a form request which would cause a page reload anyways and the user would see the messages then.